scipy/special/tests/test_support_alternative_backends.py

```diff
@@ -108,10 +192,25 @@ def test_support_alternative_backends(xp, func, nfo, dtype, shapes):
 
     res = nfo.wrapper(*args_xp)  # Also wrapped by lazy_xp_function
     ref = nfo.func(*args_np)  # Unwrapped ufunc
-
+    if (
+            is_torch(xp)
+            and xp_default_dtype(xp) == xp.float32
+            and "float64" not in dtypes
+    ):
+        # int64 promotes like float32 on torch with default dtype = float32
+        # cast reference if needed
+        ref = np.float32(ref)
```

Comment:
Eventually we should remove code duplication rather than just pasting this line in everywhere a reference output is computed, but I'd prefer to punt on what the best way to do this. Using duplication to postpone the decision basically.